docs/how-to-guides/upgrade-rook-ceph.md

```diff
@@ -0,0 +1,179 @@
+
+Run the following command in a new terminal window to keep an eye on the rook version as it is
+updated for all the sub-components:
+
+```bash
+watch --exec kubectl -n rook get deployments -l rook_cluster=rook -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+```
+
+```bash
+watch --exec kubectl -n rook get jobs -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \tsucceeded: "}{.status.succeeded}{"      \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+```
+
+You should see that `rook-version` slowly changes to `v1.4.6`.
+
+#### Step 2.4: Ceph version update
+
+Run the following command to keep an eye on the Ceph version update as the new pods come up in a new
+terminal window:
+
+```bash
+watch --exec kubectl -n rook get deployments -l rook_cluster=rook -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \tceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}'
+```
+
+You should see that `ceph-version` slowly changes to `15.2.5`.
+
+#### Step 2.5: Events in rook namespace
+
+In a new terminal leave the following command running, to keep track of the events happening in the
+`rook` namespace. Keep an eye on the column `TYPE` of the following output and especially events
+that are not of type `Normal`.
+
+```bash
+kubectl -n rook get events -w
+```
+
+### Step 3: Dashboards
+
+Monitor various dashboards.
+
+#### Step 3.1: Ceph
+
+Open the Ceph dashboard in a browser window. Read the docs
+[here](rook-ceph-storage.md#access-the-ceph-dashboard) to access the dashboard.
+
+> **NOTE**: Accessing dashboard can be a hassle because while the components are upgrading you may
+> lose access to it multiple times.
+
+#### Step 3.2: Grafana
+
+Gain access to the Grafana dashboard as instructed
+[here](monitoring-with-prometheus-operator.md#access-grafana). And keep an eye on the dashboard
+named `Ceph - Cluster`.
+
+> **NOTE**: The data in the Grafana dashboard will always be outdated compared to the `watch ceph
+> status` running inside the toolbox pod.
+
+### Step 4: Make a note of existing image versions
+
+Make a note of the images of the pods in the rook namespace:
+
+```bash
+kubectl -n rook get pod -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"\n\t"}{.status.phase}{"\t\t"}{.spec.containers[0].image}{"\t"}{.spec.initContainers[0].image}{"\n\n"}{end}'
+```
+
+After the upgrade is complete, we can verify the output of the above command to see if the workloads
+now run updated images.
+
+### Step 5: Perform updates
+
+With everything monitored, you can start the update process now by executing the following commands:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kinvolk/lokomotive/v0.5.0/assets/charts/components/rook/templates/resources.yaml
```

Comment:
Is the idea to keep this tag updated as we release?

Reply:
We have to update the doc on each upgrade of Rook Ceph. Things may change between rook-ceph releases.